pandas/core/arrays/numpy_.py

```diff
@@ -507,6 +507,23 @@ def to_numpy(
 
         return result
 
+    def _validate_setitem_value(self, value):
+        kind = self.dtype.kind
+        if kind == "b":
+            if lib.is_bool(value):
+                return value
+
+        elif kind == "f":
+            if lib.is_integer(value) or lib.is_float(value):
+                return value
```

Comment:
these checks are correct for scalar value, not sure if validate_setitem_value needs to handle listlike?

Reply:
The TypeError still occurs if listlike values are entered in to validate causing tests to fail, even if it's supposed to work, see pandas/tests/arrays/numpy_/test_numpy.py test_ufunc_unary